test/common/common/fmt_helpers_test.cc

```diff
@@ -0,0 +1,24 @@
+#include "common/common/fmt_helpers.h"
+#include "common/common/logger.h"
+
+#include "absl/strings/string_view.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+TEST(FormatHelpersTest, Format) {
+  std::string s = "This is my string.";
+  absl::string_view my_string = s.substr(8, 9);
+  absl::string_view is = s.substr(5, 2);
```

Comment:
I suspect the ASAN issues are related to the lifetime of the std::string returned by std::string::substr here.

Consider:
  absl::string s_view = s;
  absl::string_view my_string = s_view.substr(8, 9);
  absl::string_view is = s_view.substr(5, 2);

This is a better pattern from the performance perspective as well (in addition to correctness) because you want to do the substr of the string_view.

Reply:
Makes sense. Thanks.